server/lib/object-storage.ts

```diff
@@ -0,0 +1,153 @@
+    stream.byteLength = min([ stats.size - start, PART_SIZE ])
+    const uploadPartCommand = new UploadPartCommand({
+      Bucket: bucketInfo.BUCKET_NAME,
+      Key: key,
+      UploadId: createResponse.UploadId,
+      PartNumber: partNumber,
+      Body: stream
+    })
+    const uploadResponse = await s3Client.send(uploadPartCommand)
+    parts.push({ ETag: uploadResponse.ETag, PartNumber: partNumber })
+    partNumber += 1
+  }
+  await close(fd)
+  const completeUploadCommand = new CompleteMultipartUploadCommand({
+    Bucket: bucketInfo.BUCKET_NAME,
+    Key: key,
+    UploadId: createResponse.UploadId,
+    MultipartUpload: { Parts: parts }
+  })
+  await s3Client.send(completeUploadCommand)
+  logger.debug('Completed in %d parts of file to %s/%s%s', partNumber - 1, bucketInfo.BUCKET_NAME, bucketInfo.PREFIX, file.filename)
+}
+
+export async function storeObject (file: {path: string, filename: string}, bucketInfo: BucketInfo) {
+  logger.debug('Uploading file to %s/%s%s', bucketInfo.BUCKET_NAME, bucketInfo.PREFIX, file.filename)
+  const stats = await stat(file.path)
+  // If bigger than 100 MiB we do a multipart upload
+  if (stats.size > MAX_PUT_SIZE) {
+    return await multiPartUpload(file, stats, bucketInfo)
+  }
+  const fileStream = createReadStream(file.path)
+  return await objectStoragePut({ filename: file.filename, content: fileStream, bucketInfo })
+}
+
+export async function writeObjectContents (file: {filename: string, content: string}, bucketInfo: BucketInfo) {
+  logger.debug('Writing object to %s/%s%s', bucketInfo.BUCKET_NAME, bucketInfo.PREFIX, file.filename)
+  return await objectStoragePut({ filename: file.filename, content: file.content, bucketInfo })
+}
+
+export async function removeObject (filename: string, bucketInfo: BucketInfo) {
+  const key = bucketInfo.PREFIX + filename
+  const s3Client = getS3Client()
+  const command = new DeleteObjectCommand({
+    Bucket: bucketInfo.BUCKET_NAME,
+    Key: key
+  })
+  return await s3Client.send(command)
+}
+
+export async function removePrefix (prefix: string, bucketInfo: BucketInfo) {
+  const s3Client = getS3Client()
+  const listCommand = new ListObjectsV2Command({
+    Bucket: bucketInfo.BUCKET_NAME,
+    Prefix: bucketInfo.PREFIX + prefix
+  })
+
+  const listedObjects = await s3Client.send(listCommand)
+  const deleteParams = {
+    Bucket: bucketInfo.BUCKET_NAME,
+    Delete: { Objects: [] }
+  }
+  for (const object of listedObjects.Contents) {
+    deleteParams.Delete.Objects.push({ Key: object.Key })
+  }
+  const deleteCommand = new DeleteObjectsCommand(deleteParams)
+  await s3Client.send(deleteCommand)
+
+  // Repeat if not all objects could be listed at once (limit of 1000?)
+  if (listedObjects.IsTruncated) await removePrefix(prefix, bucketInfo)
+}
+
+export function generateObjectStoreUrl (filename: string, bucketInfo: BucketInfo) {
+  return `https://${bucketInfo.BUCKET_NAME}.${CONFIG.OBJECT_STORAGE.ENDPOINT}/${bucketInfo.PREFIX}${filename}`
```

Comment:
We should also support `http://` (just take the same scheme than endpoint for example)